docs/source/caching/programmatic-ids.mdx

```diff
@@ -59,25 +52,27 @@ val objectIdGenerator = object : CacheKeyGenerator {
 }
 ```
 
-Note how this requires `id` to be present in the response. If a query does not return this field, the `CacheKey` will be inconsistent and duplication will happen.
+You can also use the current object's typename to use _different_ cache ID generation logic for different object types.
+
+Note that for cache ID generation to work, your GraphQL operations must return whatever fields your custom code relies on (such as `id` above). If a query does not return a required field, the cache ID will be inconsistent, resulting in data duplication.
 
-## CacheKeyResolver
+## `CacheKeyResolver`
 
-To customize how cache ids are built from fields arguments (before going to the network), create a new instance of `CacheKeyResolver`:
+The `CacheKeyResolver` class enables you to generate custom cache IDs from an field's arguments. This basic example generates _every_ object type's cache ID from the `id` argument _if it's present_:
 
 ```kotlin
 val cacheKeyResolver = object: CacheKeyResolver() {
   override fun cacheKeyForField(field: CompiledField, variables: Executable.Variables): CacheKey? {
-    // [field] contains compile-time information about what type of object is being resolved
+    // `field` contains compile-time information about what type of object is being resolved.
```

Comment:
FWIW, the `[identifier]` syntax is the [KDoc](https://kotlinlang.org/docs/kotlin-doc.html) way to do. Backticks work too !

Reply:
Ok cool! I'll go with backticks for now just for cross-docset consistency within comments